lib/puppet/type/logical_volume.rb

```diff
@@ -77,6 +77,15 @@
     end
   end
 
+  newparam(:readahead) do
+    desc "The readahead count to use for the new logical volume."
+    validate do |value|
+      unless value =~ /^([0-9]+|Auto|None)/i
```

Comment:
Does lvcreate care about the casing for auto or none? If it does this parameter should have a munge block to fix the case.